src/render/processor.rs

```diff
@@ -58,22 +59,24 @@ pub trait AudioProcessor: Send {
 ///
 /// Provided to implementations of [`AudioProcessor`] in the render thread
 pub struct AudioParamValues<'a> {
-    nodes: &'a FxHashMap<NodeIndex, Node>,
+    nodes: &'a FxHashMap<NodeIndex, RefCell<Node>>,
 }
 
 impl<'a> AudioParamValues<'a> {
-    pub(crate) fn from(nodes: &'a FxHashMap<NodeIndex, Node>) -> Self {
+    pub(crate) fn from(nodes: &'a FxHashMap<NodeIndex, RefCell<Node>>) -> Self {
         Self { nodes }
     }
 
     pub(crate) fn get_raw(&self, index: &AudioParamId) -> &AudioRenderQuantum {
-        self.nodes.get(&index.into()).unwrap().get_buffer()
+        unsafe { (*self.nodes.get(&index.into()).unwrap().as_ptr()).get_buffer() }
```

Comment:
Sorry, I tried, but this `unsafe` snippet is fundamentally unsound.
If I modify the function signatures to make the returned references of static lifetime, it still compiles:
```rust
-    pub(crate) fn get_raw(&self, index: &AudioParamId) -> &AudioRenderQuantum {
+    pub(crate) fn get_raw(&self, index: &AudioParamId) -> &'static AudioRenderQuantum {
         unsafe { (*self.nodes.get(&index.into()).unwrap().as_ptr()).get_buffer() }
     }

     /// Get the computed values for the given [`crate::param::AudioParam`]
     ///
     /// For both A & K-rate params, it will provide a slice of length [`crate::RENDER_QUANTUM_SIZE`]
-    pub fn get(&self, index: &AudioParamId) -> &[f32] {
+    pub fn get(&self, index: &AudioParamId) -> &'static [f32] {
         // let buffer = self.nodes.get(&index.into()).unwrap().borrow().get_buffer();
         // &buffer.channel_data(0)[..]
         &self.get_raw(index).channel_data(0)[..]
```
Which is obvious. By moving to a ptr and then back, you are erasing the lifetime information.
This is problematic. A user provided `AudioProcessor` might take the `&'static [f32]` from the `AudioParamValues.get` method, and store it for a longer time than the buffer exists (the graph may deallocate it already in the next `render` call)

Let me try to make it sound by dropping unsafe, but the function signature will suffer. Let's see if it is acceptable

Reply:
Hum, I'm not sure I understand well, these lifetime stuff are still a bit like dark magic for me... (slowly understanding things but, slowly :)

I just wonder naively, why can't we use the same trick you did in `Graph::render()` and just rely on a clone somehow?

Reply:
> these lifetime stuff are still a bit like dark magic for me

which is fine, unless you use `unsafe` to mess with lifetimes ;)

> I just wonder naively, why can't we use the same trick you did in Graph::render() and just rely on a clone somehow?

Definitely, but the function signatures will change.
One simple way would be to have the `get` method return an AudioRenderQuantumChannel
```
impl<'a> AudioParamValues<'a>

    pub fn get(&self, index: &AudioParamId) -> impl std::ops::Deref<Target = [f32; 128]> + '_ {
        self.nodes
            .get(&index.into())
            .unwrap()
            .borrow()
            .get_buffer()
            .channel_data(0)
            .clone()
    }
}
```

But now, it will deref always to a `[f32; 128]` fixed length array. In the current impl we deref to `[f32]`, a slice with no fixed length. This leaves room for your work in #196